### PR TITLE
Fix context initialization

### PR DESCRIPTION
**Pull request by atobiszei**
ID:CVS-153260

- [x] Unit tests added.
- [ ] The documentation updated.
- [x] Change follows security best practices.

### CODE REVIEW

File: src/test/modelinstance_test.cpp [hunk 1]
```diff
@@ -381,6 +382,55 @@ TEST_F(TestLoadModel, CheckIfNonExistingXmlFileReturnsFileInvalid) {
...
+            false,                 // low latency transformation enabled
+            500,                   // stateful sequence max number,
+            "",                    // cache dir
+            1,                     // version
+            dummy_model_location,  // local path
+        };
+    }
+};
+TEST_F(TestLoadModelWithRemoteTensorFactoriesSucceeds, CheckIfLoadingSucceedsForGPU) {
+    SKIP_AND_EXIT_IF_NO_GPU();
```
Comment (dkalinowski) [inline src/test/modelinstance_test.cpp:409]: We agreed to contain OpenCL in the name if test is for gpu. And VAContext if it is VA related. @ngrozae

File: src/modelinstance.cpp [hunk 2]
```diff
@@ -754,7 +754,7 @@ Status ModelInstance::loadOVModelUsingCustomLoader() {
 namespace ovms {
 void ModelInstance::loadCompiledModelPtr(const plugin_config_t& pluginConfig) {
     OV_LOGGER("ov::Core: {}, ov::Model: {}, targetDevice: {}, ieCore.compile_model(model, targetDevice, pluginConfig", reinterpret_cast<void*>(&ieCore), reinterpret_cast<void*>(this->model.get()), this->targetDevice);
-    if (this->targetDevice.find("GPU") != std::string::npos) {
+    if (startsWith(this->targetDevice, "GPU")) {
```
Comment (dkalinowski) [inline src/modelinstance.cpp:757]: is it related to a fact that it cannot work with AUTO:.../MULTI:...?

File: src/test/modelinstance_test.cpp [hunk 3]
```diff
@@ -381,6 +382,55 @@ TEST_F(TestLoadModel, CheckIfNonExistingXmlFileReturnsFileInvalid) {
...
+            false,                 // idle sequence cleanup enabled
+            false,                 // low latency transformation enabled
+            500,                   // stateful sequence max number,
+            "",                    // cache dir
+            1,                     // version
+            dummy_model_location,  // local path
+        };
+    }
+};
+TEST_F(TestLoadModelWithRemoteTensorFactoriesSucceeds, CheckIfLoadingSucceedsForGPU) {
```
Comment (atobiszei) [inline src/test/modelinstance_test.cpp:408]:
```suggestion
TEST_F(TestLoadModelWithRemoteTensorFactoriesSucceeds, OpenCL_CheckIfLoadingSucceedsForGPU) {
```

File: src/test/modelinstance_test.cpp [hunk 4]
```diff
@@ -381,6 +382,55 @@ TEST_F(TestLoadModel, CheckIfNonExistingXmlFileReturnsFileInvalid) {
...
+    SKIP_AND_EXIT_IF_NO_GPU();
+    this->config.setTargetDevice("GPU");
+    ovms::ModelInstance modelInstance("UNUSED_NAME", UNUSED_MODEL_VERSION, *ieCore);
+    auto status = modelInstance.loadModel(config);
+    EXPECT_EQ(status, ovms::StatusCode::OK) << status.string();
+}
+TEST_F(TestLoadModelWithRemoteTensorFactoriesSucceeds, CheckIfLoadingSucceedsForHeteroCPUGPU) {
```
Comment (atobiszei) [inline src/test/modelinstance_test.cpp:415]:
```suggestion
TEST_F(TestLoadModelWithRemoteTensorFactoriesSucceeds, OpenCL_CheckIfLoadingSucceedsForHeteroCPUGPU) {
```

**Review by dkalinowski — APPROVED** [review]

**Review by rasapala — APPROVED** [review]

**Pull request merged**